tanglegram/tangle.py

```diff
@@ -44,83 +44,30 @@
     sh.setFormatter(formatter)
     module_logger.addHandler(sh)
 
-
-def plot(a, b, labelsA=None, labelsB=None, sort=True,
-         color_by_diff=True, link_kwargs={}, dend_kwargs={}, sort_kwargs={}):
-    """Plot a tanglegram from two dendrograms.
-
-    Parameters
-    ----------
-    (a,b) :                 pandas.DataFrame | scipy.cluster.hierarchy.linkage
-                            Dendrograms to be compared. If DataFrame, will be
-                            considered a distance matrix and linkage is
-                            generated (see ``link_kwargs``).
-    (labelsA,labelsB) :     list of str
-                            If not provided and a/b are pandas Dataframes,
-                            will try to extract from columns.
-    sort :                  bool | "random" | "step1side" | "step2side" | "permuations"
-                            If True, will try rearranging dendrogram to
-                            optimise pairing of similar values. You can provide
-                            the exact method to use as a string. ``True``
-                            defaults to "random". See ``untangle()`` for a
-                            full description.
-    link_kwargs :           dict, optional
-                            Keyword arguments to be passed to ``scipy.cluster.hierarchy.linkage``
-    dend_kwargs :           dict, optional
-                            Keyword arguments to be passed to ``scipy.cluster.hierarchy.dendrogram``
-    sort_kwargs :           dict, optional
-                            Keyword arguments to be passed to ``tanglegram.untangle``
-
-    Returns
-    -------
-    matplotlib figure
-
-    """
-    plt.style.use('ggplot')
-
-    if isinstance(a, pd.DataFrame):
-        module_logger.info('Generating linkage from distances')
-        link1 = sclust.hierarchy.linkage(sdist.squareform(a, checks=False), **link_kwargs)
-        if not labelsA:
-            labelsA = a.columns.tolist()
-    elif isinstance(a, np.ndarray):
-        link1 = a
-    else:
-        raise TypeError('Parameter `a` needs to be either pandas DataFrame or numpy array')
-
-    if isinstance(b, pd.DataFrame):
-        module_logger.info('Generating linkage from distances')
-        link2 = sclust.hierarchy.linkage(sdist.squareform(b, checks=False), **link_kwargs)
-        if not labelsB:
-            labelsB = b.columns.tolist()
-    elif isinstance(b, np.ndarray):
-        link2 = b
-    else:
-        raise TypeError('Parameter `b` needs to be either pandas DataFrame or numpy array')
-
-    if sort:
-        if not isinstance(sort, str):
-            sort = 'random'
-        link1, link2 = untangle(link1, link2,
-                                labelsA, labelsB,
-                                method=sort, **sort_kwargs)
-
-    fig = pylab.figure(figsize=(8, 8))
+def draw_tanglegram(linkage_1, linkage_2, labels1, labels2, color_by_diff=True, dend_kwargs={}):
```

Comment:
Any reason why you dropped the entire docstring and a couple parameters?

Reply:
Also am I correct in that you want to change the workflow such that people produce the linkage themselves (i.e. no more DataFrames), untangle it and then pass it to the plotting function?

Reply:
Hi. Sorry for dropping some parameters that change the use of the library. I deleted them because I did not use them, but you are right, there should be other options for users.

Yes. The workflow I wanted is that the users produce the linkages first and then use the untangle methods and "draw" function to get the desired tanglegram layout.